woocommerce/woocommerce-ios
# Fetch & Store the new storeID property.

Part of #11149

# Why

This PR takes care of fetching and storing the newly added `storeID` property from the `SystemStatus` API.

On the next PR, the `storeID` will be used to be sent on every track event.

# How

- Updated the `SystemStatusRemote` type to properly fetch the `storeID` from the environment object.

- Refactored the `SystemStatusStore` to not only fetch and store system plugins but also the `storeID`.

- Uses the `GeneralStoreSettings` type to save the new `storeID` value.  I previously attempted to store it inside the `Site` entity but the logic started to get quite complex and difficult to maintain(especially around bidirectional updates) as the `Site` APIs are not related to that new value.

# Testing Steps

**Note:** As https://github.com/woocommerce/woocommerce/pull/41341 has not been merged yet, we need to create a Jurassic Ninja store using the `add/store-id-to-telemetry-endpoint` branch.

- Create a Jurassic ninja store
- Run the app and log into that store
- See that a message similar to `🚨 Store ID: FGH58G-FGHJI-4567GH` is printed in the console

-----

- Log in to a different store (That runs a stable version of WC and does not has the store id field)
- See that a message similar to `🚨 Store ID: nil` is printed in the console


---
